pkg/validate/dynamic/dynamic.go

```diff
@@ -435,41 +435,61 @@ func (dv *dynamic) validateActionsByOID(ctx context.Context, r *azure.Resource,
 	defer cancel()
 
 	c := closure{dv: dv, ctx: ctx, resource: r, actions: actions, oid: oid}
+	if err := c.checkAccessAuthReqPayloadData(); err != nil {
+		return err
+	}
 
 	return wait.PollImmediateUntil(30*time.Second, c.usingCheckAccessV2, timeoutCtx.Done())
 }
 
 // closure is the closure used in PollImmediateUntil's ConditionalFunc
 type closure struct {
-	dv       *dynamic
-	ctx      context.Context
-	resource *azure.Resource
-	actions  []string
-	oid      *string
+	dv                *dynamic
+	ctx               context.Context
+	resource          *azure.Resource
+	actions           []string
+	oid               *string
+	subjectAttributes client.SubjectAttributes
+}
+
+func (c *closure) checkAccessAuthReqPayloadData() error {
+	c.dv.log.Info("Prepare payload for CheckAccessV2 AuthZ")
+	scope := c.dv.azEnv.ResourceManagerEndpoint + "/.default"
+	t, err := c.dv.checkAccessSubjectInfoCred.GetToken(c.ctx, policy.TokenRequestOptions{Scopes: []string{scope}})
+	if err != nil {
+		c.dv.log.Error("Unable to get the token from AAD: ", err)
+		return err
+	}
+
+	tokenClaims, err := token.ExtractClaims(t.Token)
+	if err != nil {
+		c.dv.log.Error("Unable to parse the token oid claim: ", err)
+		return err
+	}
+	c.subjectAttributes.ObjectId = tokenClaims.ObjectId
+
+	if tokenClaims.ClaimNames != nil && len(tokenClaims.Groups) == 0 {
+		c.subjectAttributes.ClaimName = client.GroupExpansion
```

Comment:
I was expecting to see the helpers for parsing the JWT, extracting groups, walking the graph API when necessary, etc, in the SDK, otherwise every client will need this logic to correctly call it, right?

Reply:
Moved this logic to SDK